### ISSUE TITLE
Build Plate Adhesion - Per Object   / Brim on/off per object

### ISSUE THREAD

**Issue by PhilBaz** [smartavionics/Cura#151]
**Is your feature request related to a problem? Please describe.**
(A clear and concise description of what the problem is. Ex. I'm always frustrated when [...])

I'm always frustrated when I have a plate of prints with a few small object that will require a brim. But most won't. Right now the best option is to use brim with a large offset distance so most parts will have a disconnected brim, then use per-object setting to remove the offset for the small pieces, so their brim is connected.

**Describe the solution you'd like**
(A clear and concise description of what you want to happen. If possible, describe why you think this is a good solution.)

Brim on/off per object settings, or build plate adhesion per object settings. The main idea is to reduce the waste of time and material associated with all the disconnected brims for large objects that don't actually need a brim.   The best would be  build plate adhesion per object settings so i could have a skirt for the objects that don't need a brim, and a brim for those that do. Though I admit i'm having trouble thinking of how that would work. Next best (and i can picture how it would work),  Brim line count could be set to zero (or 1) in main settings and >1 in per object. 

**Describe alternatives you've considered**
(A clear and concise description of any alternative solutions or features you've considered. Again, if possible, think about why these alternatives are not working out.)

Described above, just non-optimal and a tad wasteful.

**Affected users and/or printers**
(Who do you think will benefit from this? Is everyone going to benefit from these changes? Or specific kinds of users?)

Yeah, this seems like something many should benefit from. If you're printing plates of parts you know the troubles i've seen. Oh lord the troubles I've seen.

**Additional context**
(Add any other context or screenshots about the feature request here.)

I put this request on the main branch before finding your much more awesome branch.....they essentially said it was a very complex problem to address and thought the ROI was not worth it. Fair enough and I could kinda imagine why.  

Thanks again for all your efforts!

**Comment by smartavionics** [comment]
> I put this request on the main branch before finding your much more awesome branch.....they essentially said it was a very complex problem to address and thought the ROI was not worth it. Fair enough and I could kinda imagine why.

I agree, it's difficult to implement. I'll have a think about it though and maybe I will come up with something.
